### ISSUE TITLE
Tests have failed on Python 3.6 for a while, but have been hidden by misconfiguration

### ISSUE THREAD

**Issue by EmilStenstrom** [EmilStenstrom/django-components#250]
I just found out that tox-gh-actions has silently not ran any tests for Python 3.6 for a while. It seems it doesn't find the right environment and then just quits and reports success. 

Running "tox" locally confirms that django-components is currently broken in Python 3.6. Annoying. 

I've played around with CI settings, and tests now fail when no environment is found, which I think is a more truthful state to be in, but of course annoying to have failing tests.

I've asked [a question in the tox-gh-actions repo](https://github.com/ymyzk/tox-gh-actions/issues/146) to see if there's something obvious I've done incorrectly.

**Comment by EmilStenstrom** [comment]
That one test is failing isn't a fuckup, it's super easy to miss version differences like that. I've sometimes found that .tox directories retain the old versions of files when you change them. Sometimes deleting the whole .tox dir has been what's needed...

As for why "no environments found" errors show up, we'll just have to wait and see if the tox-gh-actions author has the time to troubleshoot, I really see no difference in config between 3.6 and 3.7...

**Comment by EmilStenstrom** [comment]
YES! I tried using ChatGPT and it actually found the error. We now have the correct incantations in the tox.ini file, and the tests for Python 3.6 fails again 🥳 

@lemontheme mind taking a look? Feels wrong bypassing the tests for all the version upgrades.

**Comment by EmilStenstrom** [comment]
This seems to be The Way™️: https://github.com/altdesktop/python-dbus-next/commit/4bcf72ad94f01624a2412c2e99267dfc4b5ca92f

**Comment by lemontheme** [comment]
Whoo-hoo! Test failure party. 🪩 ChatGPT can come too, even if it's going to steal all our jobs. =p

On my side, I've figured out why tox wasn't complaining locally. Turns out pyenv had failed to install several older Python versions, and instead of failing loudly upon encountering a missing Python version, tox just kind of... skipped it. 🤷‍♀️

Removing the 'annotations' future feature will indeed fix the incompatibility problem. Incorporating and testing as I type. Contrary to what I'd feared, the impact on the utility of having type hints in the first place is minimal. As long as the corresponding types are imported within a `typing.TYPE_CHECKING` guard  (as is currently the case), the IDE/type checker is still able to resolve the actual types. And to my immense relief, TYPE_CHECKING has been available since Python 3.5. 

---

Extra info for future reference:

The reason this was included in the first place was because modules _component.py_ and _component_tags.py_ each uses types defined in the other module in their function signatures. Importing these types directly doesn't work because there is a circular relationship. It's also unnecessary, seeing as neither module does anything with the required types except use them as type hints. But those types still need to come from somewhere when the type hint is evaluated at runtime. The 'annotations' future causes type hints to be evaluated as strings rather than actual types, which means they don't actually need to be imported at runtime – only when type checking. In order to make them available at type checking time, we wrap the imports in a `if typing.TYPE_CHECKING` guard.

**Comment by lemontheme** [comment]
I've been having the hardest time getting the tox test to run for the py36 environment. Tried so many things but I kept getting the strange codec error below.

```
 UnicodeDecodeError: 'ascii' codec can't decode byte 0xc3 in position 7120: ordinal not in range(128)
```

Now, this is py36 on a M1 Macbook, which means only 3.6.15 is available via pyenv (as opposed to 3.6.9, as required by the README). Still... strange error.

Know what I did? Inspired by your last message, I copy-pasted the entire traceback into ChatGPT and it actually spotted the issue. It's something I completely overlooked. Here's what it generated:

```
The line `os.path.join(os.path.dirname(__file__), "README.md")` in the `setup.py` file of the package being installed is likely the culprit, as it is trying to read a file using the default ASCII codec, which fails when encountering non-ASCII characters.

To fix this, you can try adding an encoding parameter to the `open` function call that reads the `README.md` file. For example:

\```python
with open(os.path.join(os.path.dirname(__file__), "README.md"), encoding='utf-8') as f:
    long_description = f.read()
\```
```

So, I added `encoding='utf-8'` in the README and now it works.

I feel truly humbled.
